docs/enhancements/1-appliedwork/README.md

```diff
@@ -0,0 +1,158 @@
+// The name of the appliedwork must be the same as {work name}
+// The namespace of the appliedwork should be the same as the resource applied on
+// the managed cluster.
+type AppliedWork struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec represents the desired configuration of AppliedWork.
+	Spec AppliedWorkSpec `json:"spec"`
+
+	// Status represents the current status of AppliedWork.
+	Status AppliedtWorkStatus `json:"status,omitempty"`
+}
+
+// AppliedWorkList contains a list of AppliedWork
+type AppliedWorkList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard list metadata.
+	metav1.ListMeta `json:"metadata,omitempty"`
+	// List of works.
+	Items []AppliedWork `json:"items"`
+}
+
+// AppliedWorkSpec represents the desired configuration of AppliedWork
+type AppliedWorkSpec struct {
+	// WorkName represents the name of the related work on the hub.
+	WorkName string `json:"workName"`
+}
+
+// AppliedtWorkStatus represents the current status of AppliedWork
+type AppliedtWorkStatus struct {
+	// AppliedResources represents a list of resources defined within the work that are applied.
+	// Only resources with valid GroupVersionResource, namespace, and name are suitable.
+	// An item in this slice is deleted when there is no mapped manifest in work.Spec or by finalizer.
+	// The resource relating to the item will also be removed from managed cluster.
+	// The deleted resource may still be present until the finalizers for that resource are finished.
+	// However, the resource will not be undeleted, so it can be removed from this list and eventual consistency is preserved.
+	AppliedResources []AppliedManifestResourceMeta `json:"appliedResources,omitempty"`
+}
+
+// AppliedManifestResourceMeta represents the group, version, resource, name and namespace of a resource.
+// Since these resources have been created, they must have valid group, version, resource, namespace, and name.
+type AppliedManifestResourceMeta struct {
+	// Group is the API Group of the Kubernetes resource.
+	Group string `json:"group"`
+
+	// Version is the version of the Kubernetes resource.
+	Version string `json:"version"`
+
+	// Resource is the resource name of the Kubernetes resource.
+	Resource string `json:"resource"`
+
+	// Name is the name of the Kubernetes resource.
+	Name string `json:"name"`
+
+	// UID is set on successful deletion of the Kubernetes resource by controller. The
+	// resource might be still visible on the managed cluster after this field is set.
+	// It is not directly settable by a client.
+	UID types.UID `json:"uid,omitempty"`
+}
+```
+
+An example of `AppliedWork`:
+
+```yaml
+apiVersion: multicluster.x-k8s.io/v1alpha1
+kind: AppliedWork
+metadata:
+  name: applied-mw-01
+  namespace: default
+spec:
+  workName: mw-01
+status:
+  appliedResources:
+  - group: ""
+    name: hello
+    resource: pods
+    uid: b132cade-0500-4188-9fff-5996ca231aa3
+    version: v1
```

Comment:
one thing we should mention is it will have a ownerref to any other resources. So when applied work is deleted, or related applied resource will be cascade deleted.